src/app/src/androidTest/java/br/com/italoromeiro/arctouchtest/robots/SearchRobot.java

```diff
@@ -0,0 +1,24 @@
+package br.com.italoromeiro.arctouchtest.robots;
+
+import br.com.italoromeiro.arctouchtest.R;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+/**
+ * Created by italo on 13/08/16.
+ */
+public class SearchRobot {
+    public SearchRobot where(String nameOfStreet) {
```

Comment:
The concept behind that approach is that we need to separate the "what" from the "how" when we do tests.

You can imagine ourselves like a QA guy seeing a screen. The way he solves the problem of test a view is thinking like a human that he is. In other words, thinking about our screen, we can say that:

"To test this screen I will type WHERE I want to go. Then I will click SEARCH and after that I will expect a RESULT from that search about my place of interest."

I agree with that I can change the name from "where" to "whereToGo". It sounds better. But I will not merge the "where" and "search" methods because I will reuse them in different scenarios. Today we have just a "where" place, but tomorrow who knows? I'd write more tests to show a better use of it, but the time to test was over.

Finally, this approach is based on that lecture from Jack Wharton:

http://jakewharton.com/testing-robots/

ps: I don't know if you knows Jack Wharton, so:
https://github.com/JakeWharton
http://jakewharton.com/